py/modsys.c

```diff
@@ -147,6 +161,65 @@ STATIC mp_obj_t mp_sys_getsizeof(mp_obj_t obj) {
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(mp_sys_getsizeof_obj, mp_sys_getsizeof);
 #endif
 
+#if MICROPY_PY_SYS_PROFILING
+// prof_mode([mode]): Optionaly set the profiling mode and return the previous.
+STATIC mp_obj_t mp_sys_prof_mode(size_t n_args, const mp_obj_t *args) {
+
+    mp_obj_t prev = MP_OBJ_NEW_SMALL_INT(MP_STATE_VM(prof_mode));
+    if (n_args > 0 && MP_OBJ_IS_INT(args[0])) {
+        MP_STATE_VM(prof_mode) = mp_obj_get_int(args[0]);
+    }
+
+    return prev;
+}
+MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(mp_sys_prof_mode_obj, 0, 1, mp_sys_prof_mode);
+
+// settrace(tracefunc): Set the system’s trace function.
+STATIC mp_obj_t mp_sys_settrace(mp_obj_t obj) {
+
+    if (mp_obj_is_callable(obj)) {
+        MP_STATE_THREAD(prof_instr_tick_callback) = obj;
```

Comment:
How this variable got named "instr_**tick**_callback", if it appears in function named "settrace"? ;-)